### PR TITLE
HDF5: Explicit control over chunking

### PR DESCRIPTION
**Pull request by franzpoeschel**
```cpp
Series("output.h5", Access::CREATE, R"(hdf5.dataset.chunks = "auto")");
// ...
Dataset dataset = Dataset(datatype, global_extent, R"(
[hdf5.dataset]
chunks = [10, 100]
)");
record_component.resetDataset(dataset);
```

TODO:

- [x] Merge #1580 first
- [x] Documentation
- [x] Further testing
- [x] Decide if the JSON helper should be exposed publicly

### CODE REVIEW

**Comment by guj** [comment]
I see there are two styles of json  for series and dataset
- hdf5.dataset.chunks = "auto" 
- [hdf5.dataset] chunks = [10, 100]

any chance to consolidate?

**Comment by franzpoeschel** [comment]
> I see there are two styles of json for series and dataset
> 
>     * hdf5.dataset.chunks = "auto"
> 
>     * [hdf5.dataset] chunks = [10, 100]
> 
> 
> any chance to consolidate?

That is TOML, and both forms are equivalent in TOML. The JSON form would be `{"hdf5":{"dataset":{"chunks": "..."}}}`.
[TOML](https://toml.io/en/) does not have a canonical form on purpose, you can write either of the above as the other as well.

**Comment by guj** [comment]
Oh, I see. Was wondering. 
 
> That is TOML, and both forms are equivalent in TOML. The JSON form would be `{"hdf5":{"dataset":{"chunks": "..."}}}`. [TOML](https://toml.io/en/) does not have a canonical form on purpose, you can write either of the above as the other as well.

File: src/IO/HDF5/HDF5IOHandler.cpp [hunk 1]
```diff
@@ -579,24 +664,50 @@ void HDF5IOHandlerImpl::createDataset(
...
         {
-            //! @todo add per dataset chunk control from JSON config
-
-            // get chunking dimensions
-            std::vector<hsize_t> chunk_dims =
-                getOptimalChunkDims(dims, toBytes(d));
-
             //! @todo allow overwrite with user-provided chunk size
             // for( auto const& val : parameters.chunkSize )
             //    chunk_dims.push_back(static_cast< hsize_t >(val));
```
Comment (ax3l) [inline src/IO/HDF5/HDF5IOHandler.cpp]: @franzpoeschel can this todo now be removed?

**Review by ax3l — APPROVED** [review]
Looks great!

Please see last inline comment and then feel free to merge :)

File: src/IO/HDF5/HDF5IOHandler.cpp (hunk 1 above)
Comment (franzpoeschel) [reply src/IO/HDF5/HDF5IOHandler.cpp]: yep, this todo is now fixed

**Comment by franzpoeschel** [comment]
> Looks great!
> 
> Please see last inline comment and then feel free to merge :)

Thanks! I removed the old todos

**Pull request merged**